File: Tutoriels/Java/Projets/code/GProject/src/manager/GConfig.java
//===============================================
package manager;
//===============================================
import java.util.HashMap;
import java.util.Map;
//===============================================
public class GConfig {
	//===============================================
	private static GConfig m_instance = null; 
	Map<String, String> m_dataMap;
	//===============================================
	public GConfig() { 
		m_dataMap = new HashMap<String, String>();
	} 	
	//===============================================
	public static GConfig Instance() { 
		if(m_instance == null) { 
			m_instance = new GConfig(); 
		}
		return m_instance; 
	} 
	//===============================================
	public void setData(String key, String data) { 
        m_dataMap.put(key, data);
    } 
	//===============================================
	public String getData(String key) { 
        return m_dataMap.get(key);
    } 
	//===============================================
	public void showData() { 
		for(String key: m_dataMap.keySet()) {
            System.out.print(key + " = " + m_dataMap.get(key) + "\n");
		}
	} 
	//===============================================
}
//===============================================

File: Tutoriels/Cpp/Patrons_Conception/code/src/manager/GConfig.cpp
//================================================
#include "GConfig.h"
//================================================
GConfig* GConfig::m_instance = 0;
//================================================
GConfig::GConfig() {

}
//================================================
GConfig::~GConfig() {
	
}
//================================================
GConfig* GConfig::Instance() {
	if(m_instance == 0) {
		m_instance = new GConfig;
	}
	return m_instance;
}
//================================================
void GConfig::setData(const string& key, const string& data) {
    m_dataMap[key] = data;
}
//================================================
string GConfig::getData(const string& key) {
    return m_dataMap[key];
}
//================================================
void GConfig::showData() {
    for( map<string, string>::iterator lData = m_dataMap.begin(); lData != m_dataMap.end(); lData++) {
        std::cout << lData->first << " = " << lData->second << "\n";
    }
    cout << "\n";
}
//================================================
